runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -105,13 +107,20 @@ public void write(IToken token, Object data)
             l.getReplicationMode(token.getTokenValue())
                         .getReplicationProtocol(runtime)
                         .write(l, ld);
-
             return null;
          });
 
-
         // Cache the successful write
         if (!runtime.isCacheDisabled()) {
+            if (ld.hasCheckpointMetadata()) {
```

Comment:
If I don't do it, I get this explosion of NPEs.  The cached entry needs fixing up, here or elsewhere, prior to use after pulling out of the cache.

    smoke1Test                                                  [FAIL - RuntimeException:L101]
    smoke2Test                                                  [FAIL - RuntimeException:L176]
    checkpointWriterInterleavedTest                             [FAIL - RuntimeException:L350]
    checkpointWriterTest                                        [FAIL - RuntimeException:L237]
    Tests run: 4, Failures: 0, Errors: 4, Skipped: 0, Time elapsed: 2.092 sec <<< FAILURE! - in org.corfudb.runtime.checkpoint.CheckpointSmokeTest
    smoke1Test(org.corfudb.runtime.checkpoint.CheckpointSmokeTest)  Time elapsed: 1.06 sec  <<< ERROR!
    java.lang.RuntimeException: java.lang.NullPointerException
    	at org.corfudb.runtime.object.VersionLockedObject.lambda$syncStreamUnsafe$3(VersionLockedObject.java:532)
    	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
    	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
    	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:580)
    	at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:270)
    	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
    	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
    	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
    	at org.corfudb.runtime.view.stream.StreamSpliterator.tryAdvance(StreamSpliterator.java:63)
    	at java.util.Spliterator.forEachRemaining(Spliterator.java:326)
    	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
    	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
    	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
    	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
    	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
    	at java.util.stream.ReferencePipeline.forEachOrdered(ReferencePipeline.java:423)
    	at org.corfudb.runtime.object.VersionLockedObject.syncStreamUnsafe(VersionLockedObject.java:517)
    	at org.corfudb.runtime.object.VersionLockedObject.syncObjectUnsafe(VersionLockedObject.java:284)
    	at org.corfudb.runtime.object.CorfuCompileProxy.lambda$accessInner$1(CorfuCompileProxy.java:137)
    	at org.corfudb.runtime.object.VersionLockedObject.access(VersionLockedObject.java:184)
    	at org.corfudb.runtime.object.CorfuCompileProxy.accessInner(CorfuCompileProxy.java:136)
    	at org.corfudb.runtime.object.CorfuCompileProxy.access(CorfuCompileProxy.java:118)
    	at org.corfudb.runtime.collections.SMRMap$CORFUSMR.get(SMRMap$CORFUSMR.java:183)
    	at org.corfudb.runtime.checkpoint.CheckpointSmokeTest.smoke1Test(CheckpointSmokeTest.java:101)
    	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    	at java.lang.reflect.Method.invoke(Method.java:498)
    	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
    	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
    	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
    	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
    	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
    	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
    	at org.corfudb.AbstractCorfuTest$1$1.evaluate(AbstractCorfuTest.java:76)
    	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
    	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
    	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
    	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
    	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
    	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
    	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
    	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
    	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
    	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
    	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:367)
    	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:274)
    	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
    	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:161)
    	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:290)
    	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:242)
    	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:121)
    Caused by: java.lang.NullPointerException
    	at org.corfudb.runtime.object.VersionLockedObject.lambda$syncStreamUnsafe$3(VersionLockedObject.java:523)
    	... 51 more
    
    Results :
    
    Tests in error: 
      CheckpointSmokeTest.checkpointWriterInterleavedTest:350 » Runtime java.lang.Nu...
      CheckpointSmokeTest.checkpointWriterTest:237 » Runtime java.lang.NullPointerEx...
      CheckpointSmokeTest.smoke1Test:101 » Runtime java.lang.NullPointerException
      CheckpointSmokeTest.smoke2Test:176->lambda$smoke2Test$0:151 » Runtime java.lan...
    
    Tests run: 4, Failures: 0, Errors: 4, Skipped: 0